torch/_inductor/config.py

```diff
@@ -421,6 +421,16 @@ def decide_compile_threads():
 # Pad input tensors of matmul/bmm/addmm to leverage Tensor Cores in NVIDIA GPUs
 shape_padding = os.environ.get("TORCHINDUCTOR_SHAPE_PADDING", "1") == "1"
 
+# Control if we will do padding for pointwise/reductions
+comprehensive_padding = (
+    os.environ.get("TORCHINDUCTOR_COMPREHENSIVE_PADDING", "1") == "1"
+)
+pad_channels_last = False
+
+# Whether to treat output of the backward graph as user visible.
+# For user visible outputs, inductor will make sure the stride matches with eager.
+bw_outputs_user_visible = True
```

Comment:
Can you say more why you needed to do this ? I saw alban comment something along the lines of gradient stealing or something?

Reply:
Previously inductor does not guarantee that the output of backward graph (e.g. gradients for parameters or graph inputs if they require gradients) have the same strides as eager. This may ends up with the case that a gradient has a different layout compared to the corresponding parameter. The PyTorch autograd engine has something called gradient layout contract that may do a extra copy in that case so that the strides of gradient and parameter match. This makes the optimizer more efficient. Treating output of backward graph user visible can avoid that extra copy.

On the other hand, it's just good to make the strides of gradients matching eager in case user runs some view op on them.